test/services/CollectionServiceTest.scala

```diff
@@ -0,0 +1,74 @@
+package services
+
+import com.gu.facia.client.models.{CollectionConfigJson, CollectionJson, ConfigJson, FrontJson, Trail}
+import conf.ApplicationConfiguration
+import org.joda.time.DateTime
+import org.scalatest.{DoNotDiscover, FreeSpec, Matchers}
+import play.api.{Configuration}
+import slices.{Containers, FixedContainers, Story}
+
+@DoNotDiscover class CollectionServiceTest extends FreeSpec with Matchers {
+  val configuration = Configuration("example" -> "config")
+  val config = new ApplicationConfiguration(configuration, false)
+  val fixedContainers = new FixedContainers(config)
+  val containers = new Containers(fixedContainers)
+  val containerService = new ContainerService(containers)
+
+  "getArticleDetailsForCollection" - {
+    "should return article details for a given collection" in {
+      CollectionService.getStoriesVisibleByStage("id", collectionJson, configJson, containerService) should be(
+        getStoriesVisible(1,1,2,2)
+      )
+    }
+  }
+
+  private def getStoriesVisible(liveDesktop: Int, liveMobile: Int, draftDesktop: Int, draftMobile: Int) = {
+    Some(
+      StoriesVisibleByStage(
+        Some(
+          StoriesVisibleResponse(
+            Some(liveDesktop),
+            Some(liveMobile)
+          )),
+        Some(StoriesVisibleResponse(
+          Some(draftDesktop),
+          Some(draftMobile)
+        ))
+      )
+    )
+  }
+
+  private def collectionJson: CollectionJson = {
+    val live = List(Trail("existingId", 0, Some(""), None))
+    val draft = Trail("newId", 0, Some(""), None) :: live
+    CollectionJson(live, Some(draft), None, new DateTime(0), "oldUpdatedBy", "oldUpdatedEmail", None, None, None)
+  }
+
+  private def configJson: ConfigJson = {
+    ConfigJson(Map[String, FrontJson](), Map[String, CollectionConfigJson]("id" -> collectionConfig))
+  }
+
+  private def collectionConfig: CollectionConfigJson = {
+    CollectionConfigJson(
```

Comment:
do we do this often? that is, construct a largely empty `CollectionConfigJson`? If so, we could push its creation to the companion object to make things a bit DRYer.